File: Prototype/BigGp/src/Model/SmallNodes/Operators/Arithmetic/MultiplicativeOperator.java
package Model.SmallNodes.Operators.Arithmetic;

import Model.Node;
import Model.SmallNodes.Expressions.Standart.TwoArgExpression;
import Model.SmallNodes.Operators.Operator;
import Serialization.Token;

import java.util.ArrayList;


public class MultiplicativeOperator extends Operator {

    @Override
    public void mutate() {
        if (this.getParentNode() instanceof TwoArgExpression twoArgExpression){
            twoArgExpression.mutate();
        }
    }

    public MultiplicativeOperator(Node parentNode) {
        super(parentNode, "MultiplicativeOperator");

        if (getRandomInt(2) == 0){
            setOperator(" * ");
        } else {
            setOperator(" / ");
        }
    }

    public MultiplicativeOperator(Node parentNode, ArrayList<Token> tokens) {
        super(parentNode, tokens);
    }
}

File: Interactive.Genetic.Programming.Environment/src/Domain/Model/Nodes.Small/Operators.Arithmetic/MultiplicativeOperator.cs
using Model.Abstract;
using Utils;

namespace Model.Nodes.Small.Operators.Arithmetic;

public sealed class MultiplicativeOperator : Operator
{
    public override void Mutate()
    {
        if (ParentNode is TwoArgExpression twoArgExpression)
        {
            twoArgExpression.Mutate();
        }
    }

    public MultiplicativeOperator(Node parentNode) : base(parentNode, "MultiplicativeOperator")
    {
        Value = RandomService.RandomPercentage() < 50
            ? " * "
            : " / ";
    }

    public MultiplicativeOperator(Node parentNode, IList<Token> tokens) : base(parentNode, tokens)
    {
    }
}